### DISCUSSION TITLE
the installation is complicated

### DISCUSSION THREAD

**Discussion by josearg150** [Synthesized-Infinity/Whatsapp-Botto-Xre#6] [General]
Could you make a video on how to install it on windows?

**Comment by SomnathDas** [comment]
Sure! I'll do that as soon as I'll be free.

**Comment by SomnathDas** [comment]
@AlenSaito1 what you think? xD how about publishing it on my dead ass channel

**Reply by AlenVelocity** [reply]
Indeed

**Comment by SomnathDas** [comment]
It's been a while but I think deploying it to a cloud if you want a direct deployment without hassle is easier and more efficient than setting it up on your computer uwu 
Here's a guide on that https://github.com/Synthesized-Infinity/Whatsapp-Botto-Xre/blob/master/Heroku_Atlas_Guide.md

**Comment by PrajjwalDatir** [comment]
> Could you make a video on how to install it on windows?

@josearg150 

**Video to deploy Xre on Heroku** at your service, Dokyusei!
https://www.youtube.com/watch?v=tsCCmxeklHw 

Better to run it on Heroku than setting it up on your windows pc.

**Reply by ghost** [reply]
I just set up the bot via Heroku and I wanted to know if it would be possible to adjust the script/code myself + also limit the bot only to answer in group chats with certain names

**Reply by PrajjwalDatir** [reply]
> I just set up the bot via Heroku and I wanted to know if it would be possible to adjust the script/code myself

Fork the repo.
Clone it to your local system.
Make changes you wish to do.
Push them on Github to your forked version.
click on deploy to heroku button on your forked repo and follow the same steps as in video.

> also limit the bot only to answer in group chats with certain names

the only way i can think of to do this, is by adding a check condition in loop of the code
@AlenSaito1 @SomnathDas might have some better way-

UwO

**Reply by ghost** [reply]
> > I just set up the bot via Heroku and I wanted to know if it would be possible to adjust the script/code myself
> 
> Fork the repo.
> Clone it to your local system.
> Make changes you wish to do.
> Push them on Github to your forked version.
> click on deploy to heroku button on your forked repo and follow the same steps as in video.
> 
> > also limit the bot only to answer in group chats with certain names
> 
> the only way i can think of to do this, is by adding a check condition in loop of the code
> @AlenSaito1 @SomnathDas might have some better way-
> 
> UwO

I can't code for shit (besides some basics in JavaEditor) so I don't even know what you're saying😅

**Comment by SomnathDas** [comment]
``` I was music```

**Comment by RICH132** [comment]
I have a problem 
_npm ERR! cipm can only install packages when your package.json and package-lock.json or npm-shrinkwrap.json are in sync. Please update your lock file with `npm install` before continuing._
how do i solve it ?

**Comment by PrajjwalDatir** [comment]
Are you running locally or are deploying to heroku? @RICH132 

@SomnathDas @AlenSaito1

**Reply by RICH132** [reply]
heroku...

**Reply by PrajjwalDatir** [reply]
> heroku...

You can watch this video to deploy.
https://www.youtube.com/watch?v=tsCCmxeklHw

Xre is not actively maintained.
So, I'll say use https://github.com/PrajjwalDatir/Kaoi this.

Join group to ask doubts, if you face any difficulties
https://chat.whatsapp.com/DyNaOeTLd5A3uifKOEyRno

**Reply by RICH132** [reply]
THX a lot..

**Comment by DarkPhantom1805** [comment]
Bro i followed your tutorial but it gave me this error. Any help?


There was an issue building your app. This can mean your app.json's project is not a valid Heroku application. Please ensure your app is deployable to Heroku and try again.

-----> Building on the Heroku-20 stack
-----> Using buildpacks:
       1. heroku/nodejs
       2. https://github.com/AlenSaito1/heroku-buildpack-imagemagick.git
       3. https://github.com/clhuang/heroku-buildpack-webp-binaries.git
       4. https://github.com/jonathanong/heroku-buildpack-ffmpeg-latest.git
-----> Node.js app detected
       
-----> Creating runtime environment
       
       NPM_CONFIG_LOGLEVEL=error
       NODE_VERBOSE=false
       NODE_ENV=production
       NODE_MODULES_CACHE=true
       
-----> Installing binaries
       engines.node (package.json):  unspecified
       engines.npm (package.json):   unspecified (use default)
       
       Resolving node version 14.x...
       Downloading and installing node 14.18.2...
       Using default npm version: 6.14.15
       
-----> Installing dependencies
       Installing node modules
       npm ERR! cipm can only install packages when your package.json and package-lock.json or npm-shrinkwrap.json are in sync. Please update your lock file with `npm install` before continuing.
       npm ERR! 
       npm ERR! 
       npm ERR! Invalid: lock file's axios@0.21.1 does not satisfy axios@>=0.21.2
       npm ERR! Invalid: lock file's mongoose@5.12.2 does not satisfy mongoose@^6.0.7
       npm ERR! Invalid: lock file's yt-search@2.7.6 does not satisfy yt-search@^2.10.1
       npm ERR! 
       
       npm ERR! A complete log of this run can be found in:
       npm ERR!     /tmp/npmcache.sh8eP/_logs/2021-12-02T14_44_04_555Z-debug.log
-----> Build failed
       
       We're sorry this build is failing! You can troubleshoot common issues here:
       https://devcenter.heroku.com/articles/troubleshooting-node-deploys
       
       Some possible problems:
       
       - Node version not specified in package.json
         https://devcenter.heroku.com/articles/nodejs-support#specifying-a-node-js-version
       
       Love,
       Heroku
       
 !     Push rejected, failed to compile Node.js app.
 !     Push failed
